pkg/bbgo/order_executor_general.go

```diff
@@ -274,22 +283,35 @@ func (e *GeneralOrderExecutor) OpenPosition(ctx context.Context, options OpenPos
 		submitOrder.Side = types.SideTypeBuy
 		submitOrder.Quantity = quantity
 
-		Notify("Opening %s long position with quantity %f at price %f", e.position.Symbol, quantity.Float64(), price.Float64())
-		createdOrder, err2 := e.SubmitOrders(ctx, submitOrder)
-		if err2 != nil {
-			return err2
+		Notify("Opening %s long position with quantity %v at price %v", e.position.Symbol, quantity, price)
```

Comment:
Call this for the first time:

createdOrder, err := e.SubmitOrders(ctx, submitOrder)

 if err != nil then we should call reduceQuantityAndSubmitOrders() method

Reply:
done